### PR TITLE
Support Java behaviour w.r.t fmin/fmax/dmin/dmax on Z

### PR DESCRIPTION
**Pull request by matthewhall2**
- Enables inlining of Math.max/Math.min functions for floats and doubles
- Implements Java standard
  - +0 compares as strictly greater than -0
  - if the first arg is a NaN, returns the NaN unchanged. If only the second arg is a NaN, returns that NaN unchanged

Uses SIMD instructions for floats on z14+, and for doubles on z13+ (fixes errors on z13 machines that were seen after https://github.com/eclipse-openj9/openj9/pull/20530). 

https://github.ibm.com/runtimes/openj9-jit-z/issues/935#issuecomment-98667018

depends on https://github.com/eclipse-omr/omr/pull/7572

[vmfarm](http://vmfarm.rtp.raleigh.ibm.com/build_info.php?build_id=82594)
[vmfarm w recognizedmethodtests](http://vmfarm.rtp.raleigh.ibm.com/build_info.php?build_id=82653)

### CODE REVIEW

**Comment by matthewhall2** [comment]
please review @r30shah

File: test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangMath.java [hunk 1]
```diff
@@ -1,33 +1,42 @@
 /*
- * Copyright IBM Corp. and others 2019
```
Comment (r30shah) [inline test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangMath.java:2]: Original copy right was correct, there was not issue with space here.

File: test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangMath.java [hunk 2]
```diff
@@ -1,33 +1,42 @@
...
+import org.testng.AssertJUnit;
+
 
+
+@Test(singleThreaded=true)
 public class TestJavaLangMath {
 
     /**
-    * Tests the constant corner cases defined by the {@link Math.sqrt} method.
+     * Tests the constant corner cases defined by the {@link Math.sqrt} method.
```
Comment (r30shah) [inline test/functional/JIT_Test/src/jit/test/recognizedMethod/TestJavaLangMath.java:39]: It should be consistent for comments. 
```
/**
 *
 *
 */
```

File: test/functional/JIT_Test/src/jit/test/recognizedMethod/TestMathUtils.java [hunk 3]
```diff
@@ -0,0 +1,292 @@
+/*
+* Copyright IBM Corp. and others 2024
```
Comment (r30shah) [inline test/functional/JIT_Test/src/jit/test/recognizedMethod/TestMathUtils.java:2]: Please fix the space here. Check the copyright in other files.

**Review by r30shah — CHANGES_REQUESTED** [review]
Minor nitpicks, changes looks ok to me. Please fix this and relaunch the build in internal farm again for sanity.

**Review by r30shah — APPROVED** [review]
LGTM, I will wait for OMR changes to propagate before launching tests.

**Comment by matthewhall2** [comment]
> Minor nitpicks, changes looks ok to me. Please fix this and relaunch the build in internal farm again for sanity.

Thanks @r30shah 
[vmfarm run](https://hyc-runtimes-jenkins.swg-devops.com/job/jvm.29.personal/34501/)

**Comment by r30shah** [comment]
@matthewhall2  Can you resolve the merge conflict? I see that your OMR changes have been propagated to openj9-omr so we can target to merge this today

**Comment by matthewhall2** [comment]
> @matthewhall2 Can you resolve the merge conflict? I see that your OMR changes have been propagated to openj9-omr so we can target to merge this today

done. Running internal sanity checks again

**Comment by r30shah** [comment]
Jenkins test sanity jdk11,jdk17,jdk21 zlinux

**Review by r30shah — APPROVED** [review]

**Comment by r30shah** [comment]
jenkins test sanity zlinux jdk11,jdk21

**Comment by matthewhall2** [comment]
still waiting on the [sanity run](http://vmfarm.rtp.raleigh.ibm.com/build_info.php?build_id=83149) @r30shah. will let you know when it finishes (the only failures right now are for x86)

**Comment by r30shah** [comment]
All the tests have been on jenkins passes. Merging this based on the the sanity run build from https://github.com/eclipse-openj9/openj9/pull/20716#issuecomment-2523891718 and https://github.com/eclipse-openj9/openj9/pull/20716#issuecomment-2515514780.

@matthewhall2  We should keep monitor OpenJ9 Jenkins build and internal builds in case something pop-up.
